vizro-core/src/vizro/models/_components/form/date_range_picker.py

```diff
@@ -0,0 +1,71 @@
+from typing import List, Literal, Optional
+
+import dash_mantine_components as dmc
+from dash import html
+
+try:
+    from pydantic.v1 import Field, validator
+except ImportError:  # pragma: no cov
+    from pydantic import Field, validator
+
+
+from vizro.models import VizroBaseModel
+from vizro.models._action._actions_chain import _action_validator_factory
+from vizro.models._components.form._form_utils import validate_date_picker_value, validate_max_date
+
+
+class DateRangePicker(VizroBaseModel):
+    """Temporal multi-selector `DateRangePicker`.
+
+    Can be provided to [`Filter`][vizro.models.Filter] or
+    [`Parameter`][vizro.models.Parameter]. Based on the underlying
+    [`dmc.DateRangePicker`](https://www.dash-mantine-components.com/components/datepicker#daterangepicker).
+
+    Args:
+        value (Optional[List[str]]):
+            Default start and end date for date picker. Must be 2 items. Defaults to `[min_date, max_date]`.
+        type (Literal["date_range_picker"]): Defaults to `"date_range_picker"`.
+        min_date (Optional[str]): Minimum possible date. Defaults to `None`.
+        max_date (Optional[str]): Maximum possible date. Defaults to `None`.
+        title (str): Title to be displayed. Defaults to `""`.
+        description (str): Description to be displayed above the component.
+        actions (List[Action]): See [`Action`][vizro.models.Action]. Defaults to `[]`.
+
+    """
+
+    type: Literal["date_range_picker"] = "date_range_picker"
+
+    min_date: Optional[str] = Field(None, description="Start date value for date range picker.")
+    max_date: Optional[str] = Field(None, description="End date value for date range picker.")
+    title: str = Field("", description="Title to be displayed.")
+    description: str = Field("", description="Description to be displayed above the component.")
+    value: Optional[List[str]] = Field(
+        [], description="Default start and end date for date picker.", min_items=2, max_items=2
+    )
+
+    actions = []
+
+    _input_property: str = "value"
+    _set_actions = _action_validator_factory("value")
+
+    # Re-used validators
+    _validate_value = validator("value", allow_reuse=True)(validate_date_picker_value)
+    _validate_max_date = validator("max_date", allow_reuse=True)(validate_max_date)
+
+    def build(self):
+        init_value = self.value or [self.min_date, self.max_date]  # type: ignore[list-item]
+        return html.Div(
+            [
+                html.P(self.title) if self.title else None,
+                dmc.DateRangePicker(
+                    id=self.id,
+                    description=self.description,
+                    minDate=self.min_date,
+                    value=init_value,
+                    allowSingleDateInRange=False,
```

Comment:
Ah, this means we could actually just have one DateRangePicker that allows for single and multi-select if we set this to True?

@antonymilne - do we still want to have two separate models? Or are we actually fine having one model that does single and multi-select? If we go for the second option, we could just have this model here and set this to True. Visually they both look a bit different:

DatePicker:
<img width="220" alt="Screenshot 2024-02-14 at 18 17 34" src="https://github.com/mckinsey/vizro/assets/90609403/279797b2-03dd-4274-85cd-4323dd8543d5">

DateRangePicker if only one date is selected:
![Screenshot 2024-02-14 at 18 16 35](https://github.com/mckinsey/vizro/assets/90609403/092fe9b3-7b56-4349-9abc-ac6ff73e46b8)

```suggestion
                    allowSingleDateInRange=False,
```

Reply:
Hmmmm this is a very good question and I'm not sure. It would be neater if we could just have one model to keep things simple, but also `DateRangePicker` with a single date selected looks a bit weird.

It also raises another question which is now that I look at it, I don't know why we have a separate `RangeSlider` and `Slider` 🤔 As far as I can tell `RangeSlider` with a single element `value = [1]` would function identically to a `Slider`. So we maybe we should have simplified this to just a single `vm.Slider` model. Did we consider this before? I can't remember. @maxschulz-COL?

@AnnMarieW do you know if there's any difference in behaviour between `RangeSlider` with a single element `value` vs. `Slider`? Is there a reason there's two separate dcc components for these? Other than it's maybe easier for people to find the right component.

Reply:
💡 We could have a single `DatePicker` model but then it returns either `DateRangePicker` or `DatePicker` depending on whether the user sets some field to `True/False` or supplies two vs. one `value`?

Reply:
Interesting question about using a single value in the Range Slider.  It seems to work, but it would be possible to change it to a range slider when updating it with two values in a callback.  So there would need to be some type of check to make sure that only a single value is returned.  I'm not sure if there would be any other unintended issues.

Reply:
I've split this question off into a new discussion so as not derail things here: #318.

Currently my feeling is that this is a good idea, curious to hear what others think:
💡 We could have a single `DatePicker` model but then it returns either `DateRangePicker` or `DatePicker` depending on whether the user sets some field to `True/False` or supplies two vs. one `value`?